cv_next/app/cv/[cvId]/components/commentSection/comment.tsx

```diff
@@ -106,159 +109,180 @@ export default function Comment({
-                  <button
-                    className="text-red-500"
-                    onClick={() => setShowAlert(true)}
-                  >
-                    <FaRegTrashCan
-                      data-tooltip-target="Trash Icon"
-                      style={{
-                        transform: "translateY(2px)",
-                      }}
-                      fontSize="1.4rem"
-                    />
-                  </button>
-                  <Tooltip id="Trash icon" color="" message="delete"></Tooltip>
-                  <span> </span>
-                </>
-              </span>
-            ) : !comment.parent_comment_Id ? (
-              <button className="text-green-500">
-                <FaComment
-                  size="1.2rem"
-                  style={{
-                    marginRight: "0.4rem",
-                    transform: "translateY(2px)",
-                  }}
-                  onClick={() =>
-                    setCommentOnCommentStatus(!commentOnCommentStatus)
-                  }
-                />
-              </button>
-            ) : null}
-            <span>
-              {comment.upvotes && comment.upvotes.includes(userId) ? (
-                <button>
-                  <AiFillLike
-                    onClick={() => setLikedCommentAction(false)}
-                    size="1.4rem"
-                    style={{
-                      transform: "translateY(2px)",
-                    }}
-                  />
-                </button>
-              ) : (
-                <button>
-                  <AiTwotoneLike
-                    size="1.4rem"
-                    color="grey"
-                    style={{
-                      transform: "translateY(2px)",
-                    }}
-                    onClick={() => setLikedCommentAction(true)}
-                  />
-                </button>
-              )}
-            </span>
-          </span>
-          <span style={{ alignSelf: "end" }}>
-            {comment.resolved === true ? (
-              // TODO: change appearence of comment based on if its resolved or not
-              <button
-                className="text-green-500"
-                onClick={() => setResolvedCommentAction(false)}
-              >
-                <GoCheckCircleFill
-                  style={{
-                    transform: "translateY(2px)",
-                  }}
-                  fontSize="1.4rem"
-                />
-              </button>
-            ) : (
-              <button
-                className="text-green-500"
-                onClick={() => setResolvedCommentAction(true)}
-              >
-                <GoCheckCircle
-                  style={{
-                    transform: "translateY(2px)",
-                  }}
-                  fontSize="1.4rem"
-                />
-              </button>
-            )}
-          </span>
+  const userVoted = comment.upvotes?.includes(userId);
+  const votingSection = userVoted ? (
```

Comment:
Can you extract all of these small components to be outside of this big component? It's perfectly fine to keep them in the same file, but put them outside of the main component's code.